connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -200,7 +201,26 @@ public Converter newConverter(String converterClassOrAlias, AbstractConfig confi
             throw new ConnectException(
                     "Failed to find any class that implements Converter and which name matches "
                             + converterClassOrAlias
-                            + ", available connectors are: "
+                            + ", available converters are: "
+                            + pluginNames(delegatingLoader.converters())
+            );
+        }
+        return config != null ? newConfiguredPlugin(config, klass) : newPlugin(klass);
+    }
+
+    public HeaderConverter newHeaderConverter(String converterClassOrAlias, AbstractConfig config) {
+        Class<? extends HeaderConverter> klass;
+        try {
+            klass = pluginClass(
+                    delegatingLoader,
+                    converterClassOrAlias,
+                    HeaderConverter.class
```

Comment:
are we continuing to add layers (and time) loading things? might be the right tradeoff now, but if we keep adding slowdown, we should consider spending time optimizing it as well.

Reply:
Yes, we are, but right now there's no other way to find plugins for `HeaderConverter` implementations (that don't also implement `Converter`). There's already [KAFKA-6503](https://issues.apache.org/jira/browse/KAFKA-6503) that I'd like to address in 1.1.

Reply:
sounds good. hadn't seen that bug, but is something i had also mentioned when we added the new plugin.path stuff. i would also very much like to see this.